### PR TITLE
Add support for Jails

### PR DESCRIPTION
**Pull request by sdwilsh**
Depends on #54

This adds some very basic support for jails:
* Their name
* If the are up/down
* Method to start the jail
* Method to stop the jail
* Method to restart the jail

To accomplish this, Jobs now make use of the subscription support that is built in, and that required some pretty extensive changes to the fake server as well.

Everything comes with tests, although I did have to disable `TestCachingMachineClosed`, as it manages to break subsequent tests in ways I do not understand.  The code under test there is very straight-forward, so I'm not too upset with this, for now.

### CODE REVIEW

**Comment by sdwilsh** [comment]
I don't plan to land this until I get the new methods to have an actual return.  Also worth noting that `vm.stop` and `vm.restart` look to have changed from when they were first implemented, and likely have the same issue (and certainly are not returning the types they say they will today).  It looks like iX is changing the spec, and I'm not sure where to subscribe to those changes yet.

**Comment by sdwilsh** [comment]
I have made a lot of progress, but spent a lot of time today trying to debug what was going on with test failures that resulted in my just commenting out a test to check the closed property on the connection.  Something is likely not getting cleaned up properly with it, but I'm not sure what.

I'm not sure I care about having that test, and my inability to debug it is likely due to my relative n00bness with python and asyncio (and possibly pytest!).

**Comment by sdwilsh** [comment]
It would seem that `pytest` is hanging in CI, but not locally.  This is almost certainly due to event order with the subscriptions and calls to `jail.start`, `jail.stop`, and `jail.restart`.

**Comment by sdwilsh** [comment]
...it helps if I commit my local changes :)

**Comment by sdwilsh** [comment]
@tobiasbp, @MatthewFlamm, I'll leave this up for about a day before I merge it in case either of you want to provide comments.  I can also try to pull this into multiple parts if you'd like.

**Comment by MatthewFlamm** [comment]
I don't have much insight into the Jails and Jobs portion.  One style comment is that this code is somewhat abstract class heavy already (not necessarily bad).  You are adding a few more.  One of which is `WebsocketMachine`, which to my naive eyes can only have one use: to be made concrete by `CachingMachine`.  This might go to your comments about splitting apart the logical changes. It is hard to follow the need for the structural changes amidst the addition of new features.

**Comment by sdwilsh** [comment]
`WebsocketMachine` exists possibly due to my n00bness with python.  It largely came about because of lines like this:
https://github.com/sdwilsh/aiotruenas-client/blob/main/aiotruenas_client/websockets/virtualmachine.py#L5

I was getting lint problems reported in vscode because `CachingMachine` is unbound in `virtualmachine.py`, which is true (I do not import it because `machine.py` imports all the other files, and that ends up creating a circular reference of imports).

**Comment by MatthewFlamm** [comment]
I'm also not extremely familiar with the typing aspects of Python, so I won't be of much help here.

**Comment by sdwilsh** [comment]
#53 pulls out the first and third commits, and I'll look to make a PR to pull out the state fetchers next.
